BinaryTree.java

```diff
@@ -0,0 +1,198 @@
+package com.katermar.assignment2;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by katermar on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private final int value;
+        private TreeNode left, right;
+
+        TreeNode(int item) {
+            value = item;
+            left = right = null;

```

Comment:
Just like multiple declarations in a single line, such chain assignments are more difficult to read and more error prone. Please avoid them.

Also in this case initialization with null is not really needed - fields have null value by default, but feel free to keep explicit initialization (with single assignment per line) if you like it. It does not hurt to be explicit.